syntax_sugar/infix.py

```diff
@@ -45,19 +47,19 @@ def __iter__(self):
     
     def __next__(self):
         if self.type == 'number':
-            if self.curr <= self.end:
+            if self.curr == self.end + self.step:
```

Comment:
If possible, I think it would be better to support `self.step` to be other numbers instead of just 1 or -1 in the future. In that case, I would prefer <= or >= instead of == for the if condition. But as the interface stands now, this works fine.